chiselFrontend/src/main/scala/chisel3/core/Aggregate.scala

```diff
@@ -370,7 +370,15 @@ abstract class Record extends Aggregate {
     elements.toIndexedSeq.reverse.map(e => eltPort(e._2)).mkString("{", ", ", "}")
   }
 
-  private[chisel3] lazy val flatten = elements.toIndexedSeq.flatMap(_._2.flatten)
+  private[core] override def typeEquivalent(that: Data): Boolean = that match {
+    case that: Record =>
+      this.getClass == that.getClass &&
```

Comment:
I think they should require the same class. I can't currently think of a use case where you want to create the same Record in everything but class. For library-like interfaces I think you would want to end up using the library's Record constructor, rather than trying to create your own record with the same elements.